src/main/java/com/airlines/dao/impl/AirplaneDaoImpl.java

```diff
@@ -0,0 +1,174 @@
+package com.airlines.dao.impl;
+
+import com.airlines.dao.AirplaneDao;
+import com.airlines.dao.DbConnector;
+import com.airlines.exception.DaoOperationException;
+import com.airlines.model.Airplane;
+import com.airlines.model.AirplaneModel;
+
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class AirplaneDaoImpl implements AirplaneDao {
+
+    private final static String INSERT_AIRPLANE_QUERY =
+            "INSERT INTO airplane (id, code_name, model, manufacture_date, capacity, flight_range, crew_id) " +
+                    "VALUES(?, ?, ?, ?, ?, ?, ?);";
+    private final static String SELECT_AIRPLANE_BY_ID_QUERY = "SELECT * FROM airplane WHERE id = ?;";
+    private final static String SELECT_ALL_AIRPLANES_QUERY = "SELECT * FROM airplane;";
+    private static final String DELETE_AIRPLANE_QUERY = "DELETE FROM airplane WHERE id = ?;";
+    private final static String UPDATE_CREW_ID_QUERY = "UPDATE airplane SET crew_id = ? WHERE id = ?;";
+    private final static String SELECT_AIRPLANES_BY_CREW_NAME_QUERY = """
+            SELECT *
+            FROM crew
+            INNER JOIN airplane
+            ON crew.id = airplane.crew_id
+            WHERE name = ?;""";
+
+    private final DbConnector dbConnector;
+
+    public AirplaneDaoImpl(DbConnector dbConnector) {
+        this.dbConnector = dbConnector;
+    }
+
+    @Override
+    public void save(Airplane airplane) {
+        Objects.requireNonNull(airplane);
+
+        try (Connection connection = dbConnector.getDbConnection();
+             PreparedStatement insertStatement = connection.prepareStatement(INSERT_AIRPLANE_QUERY)) {
+
+            insertStatement.setInt(1, airplane.getId());
+            insertStatement.setString(2, airplane.getCodeName());
+            insertStatement.setString(3, airplane.getModel().toString());
+            insertStatement.setDate(4, Date.valueOf(airplane.getManufactureDate()));
+            insertStatement.setInt(5, airplane.getCapacity());
+            insertStatement.setInt(6, airplane.getFlightRange());
+            insertStatement.setInt(7, airplane.getCrewId());
+            insertStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DaoOperationException("Error saving airplane" + airplane, e);
```

Comment:
What output of this line would be?
Does Airplane entity override toString method?

Reply:
done in https://github.com/pielena/airlines-jdbc-app/commit/1277c24daafc1469479f27cad63dd700e39ab392